### ISSUE TITLE
[FIPS] modkit bootstrap: Explicit post-install FIPS provider assertion

### ISSUE THREAD

**Issue by il10241024** [constructorfabric/gears-rust#3416]
> 🔗 **Mirrored from** [cyberfabric/cyberware-rust#1938](https://github.com/cyberfabric/cyberware-rust/issues/1938)
> Originally opened by **MikeFalcon77** on 2026-05-18T13:35:29Z

---

## Parent
Sub-issue of #3384

## Problem
`libs/modkit/src/bootstrap/crypto.rs` calls `install_default()` but never asserts that the installed provider actually reports `fips() == true`. This means a misconfigured build (e.g., wrong feature flags, wrong OS) can pass bootstrap without surfacing the FIPS mode failure.

## Action
Immediately after `install_default()` under `cfg(feature = "fips")`:

```rust
#[cfg(feature = "fips")]
{
    let provider = CryptoProvider::get_default()
        .expect("provider must be installed at this point");
    assert!(
        provider.fips(),
        "FIPS post-install witness failed: installed provider does not report fips()==true"
    );
    tracing::info!("FIPS post-install witness: OK");
}
```

This gives a single, explicit early-failure point with a clear log line, covering all downstream primitives (TLS client/server configs, JWT, sqlx DB TLS).

## Acceptance criteria
- [ ] Bootstrap under `--features fips` with a non-FIPS provider panics with a clear message
- [ ] Bootstrap under `--features fips` with the correct provider logs "FIPS post-install witness: OK"
- [ ] No assertion in non-FIPS builds

---
